### PR TITLE
Mbarjaktarovic update building apartment relationship

### PR DESCRIPTION
**Pull request by MBarjaktarovic**

### CODE REVIEW

File: web/src/main/java/eu/execom/buildingmanager/web/controller/TenantController.java [hunk 1]
```diff
@@ -65,13 +63,14 @@ public TenantController(CommandGateway commandGateway) {
...
+    String buildingId = changeNumberOfPersonsRequest.getBuildingId();
     String id = changeNumberOfPersonsRequest.getId();
     Integer numberOfPersons = changeNumberOfPersonsRequest.getNumberOfTenants();
 
-    return commandGateway.send(new ChangeNumberOfTenantsCommand(id, numberOfPersons));
+    return commandGateway.send(new ChangeNumberOfTenantsCommand(buildingId, id, numberOfPersons));
   }
 
-  @PutMapping("/apartment")
+  @PutMapping("/register-apartment")
```
Comment (stefankaticec) [inline web/src/main/java/eu/execom/buildingmanager/web/controller/TenantController.java]: Why change the name? 
Why is an apartment related to Tenant? 
Why isn't the endpoint related to building identity?

File: web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterTenantApartmentRequest.java [hunk 2]
```diff
@@ -1,11 +1,12 @@
 package eu.execom.buildingmanager.web.dto;
 
-import eu.execom.buildingmanager.domain.apartment.Apartment;
 import lombok.Value;
+import org.axonframework.commandhandling.TargetAggregateIdentifier;
 
 @Value
 public class RegisterTenantApartmentRequest {
 
+  @TargetAggregateIdentifier
```
Comment (stefankaticec) [inline web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterTenantApartmentRequest.java]: Why use this annotation in DTO?

File: web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterApartmentRequest.java [hunk 3]
```diff
@@ -10,7 +8,9 @@
 public class RegisterApartmentRequest {
 
   @NotNull
```
Comment (stefankaticec) [inline web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterApartmentRequest.java]: Not needed.

File: domain/src/main/java/eu/execom/buildingmanager/domain/building/Apartment.java [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
 
   private static final long serialVersionUID = 3232349227656067723L;
 
   @Id
+  @EntityId
   private String id;
 
-  @Column(nullable = false)
+  @ManyToOne
   private Building building;
```
Comment (stanislavgrujic) [inline domain/src/main/java/eu/execom/buildingmanager/domain/building/Apartment.java]: Actually, we shouldn't have reference to a building, just its id.

File: domain/src/main/java/eu/execom/buildingmanager/domain/building/Building.java [hunk 5]
```diff
@@ -67,4 +79,26 @@ public void on(BuildingRegisteredEvent event) {
...
   }
+
+  @CommandHandler
+  public void on(RegisterApartmentCommand command, IdGenerator idGenerator) {
+    String id = idGenerator.generateId();
+    Integer floor = command.getFloor();
+    Integer apartmentNumber = command.getApartmentNumber();
+
+    Assert.notNull(floor, "Missing floor");
+    Assert.notNull(apartmentNumber, "Missing apartment streetNumber");
```
Comment (stanislavgrujic) [inline domain/src/main/java/eu/execom/buildingmanager/domain/building/Building.java]: Is it apartment or streetNumber?

File: domain/src/main/java/eu/execom/buildingmanager/domain/building/Apartment.java (hunk 4 above)
Comment (MBarjaktarovic) [reply domain/src/main/java/eu/execom/buildingmanager/domain/building/Apartment.java]: We have that in more than one class, so we decide to create a new issue which will change that in whole project.

File: domain/src/main/java/eu/execom/buildingmanager/domain/building/Building.java (hunk 5 above)
Comment (stanislavgrujic) [reply domain/src/main/java/eu/execom/buildingmanager/domain/building/Building.java]: This is still not fixed nor clarified.

File: web/src/main/java/eu/execom/buildingmanager/web/controller/TenantController.java (hunk 1 above)
Comment (stanislavgrujic) [reply web/src/main/java/eu/execom/buildingmanager/web/controller/TenantController.java]: Not reverted nor clarified.

File: web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterApartmentRequest.java (hunk 3 above)
Comment (stanislavgrujic) [reply web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterApartmentRequest.java]: Not fixed.

File: web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterTenantApartmentRequest.java (hunk 2 above)
Comment (stanislavgrujic) [reply web/src/main/java/eu/execom/buildingmanager/web/dto/RegisterTenantApartmentRequest.java]: Please fix this.

File: domain/src/main/java/eu/execom/buildingmanager/domain/building/Expense.java [hunk 6]
```diff
@@ -21,7 +20,8 @@
 
   Status status;
 
-  public Expense(Money money, Status status) {
+  @JsonCreator
+  public Expense(@JsonProperty("money") Money money, @JsonProperty("status") Status status) {
```
[the remark this answers is not in the document]
Comment (Varadski) [reply domain/src/main/java/eu/execom/buildingmanager/domain/building/Expense.java]: Yes, it is necessary. If the constructor has only one parameter it isn't, but if it has more then all need to have that annotation. From the documentation:

`Constructor/factory method where every argument is annotated with @JsonProperty or @JacksonInject, to indicate name property to bind to`

File: pom.xml [hunk 7]
```diff
@@ -46,12 +46,6 @@
       <version>${axon.version}</version>
     </dependency>
 
-    <dependency>
-      <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-devtools</artifactId>
```
[the remark this answers is not in the document]
Comment (Varadski) [reply pom.xml]: Yes. Axon has issues with frameworks that do automatic reloading, where in some cases they reload the event classes but not the aggregates, causing a situation where Axon throws **IncompatibleAggregateException**.

File: domain/src/main/java/eu/execom/buildingmanager/domain/building/Expense.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (stanislavgrujic) [reply domain/src/main/java/eu/execom/buildingmanager/domain/building/Expense.java]: OK. Thanks for the clarification!

File: pom.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (stanislavgrujic) [reply pom.xml]: OK. 👍

**Pull request merged**